### DISCUSSION TITLE
Countdown timer

### DISCUSSION THREAD

**Discussion by danielciornei** [breezefront/community#214] [3. Integrations]
is there a countdown timer that supports breeze?
We were using Amasty Countdown , any chance of a compatibility module with this ? or maybe a refreshed breeze version of the swissup countdown timer module?

**Comment by vovayatsyuk** [comment]
You can get a working countdown timer using https://github.com/PButcher/flipdown library.

Paste this code directly into your CMS page (in backend editor):

```js
<link href="https://cdn.jsdelivr.net/npm/flipdown@0.3.2/dist/flipdown.min.css" rel="stylesheet">

<div id="flipdown" class="flipdown"></div>

<script type="module">
require(['https://cdn.jsdelivr.net/npm/flipdown@0.3.2/src/flipdown.min.js'], () => {
  var flip = new FlipDown((new Date('2025-12-31 23:59:59').getTime() / 1000)).start();

  $(document).one('breeze:destroy', () => {
    clearInterval(flip.countdown);
    $('#flipdown').empty();
  });
});
</script>
```

**Comment by JDavidVR** [comment]
@vovayatsyuk 

How/What would be the aproach in case I wan to download 'https://cdn.jsdelivr.net/npm/flipdown@0.3.2/src/flipdown.min.js' and server the js locally (using a module)?

(I've reviewed docs and it's not clear to me)

**Reply by vovayatsyuk** [reply]
**Include js from module**

Js file is not required to have component inside, you can include any custom js using the same approach: https://breezefront.com/docs/custom-module#js-components

**Include js from theme**

Take a look at how we include custom js in blank theme: https://github.com/breezefront/theme-frontend-breeze-blank/blob/master/Magento_Theme/layout/default_head_blocks.xml#L23

**Reply by vovayatsyuk** [reply]
To process these kind of elements:

```html
<div id="uniqueId" class="flipdown" data-mage-init='{"flipdown": {"timestamp": 1593561600}}'></div>
```

Your module may look like this:

- web/js/flipdown.min.js - copy of npm package
- web/js/flipdown-component.js - your component:

```js
$.widget('flipdown', {
    component: 'flipdown',
    create: function () {
        this.flip = new FlipDown(this.options.timestamp, this.element.attr('id')).start();
    },
    destroy: function() {
        clearInterval(this.flip.countdown);
        this.element.empty();
    }
});
```

breeze_default.xml:

```xml
<referenceBlock name="breeze.js">
  <arguments>
    <argument name="bundles" xsi:type="array">
      <item name="default" xsi:type="array">
        <item name="items" xsi:type="array">
          <item name="flipdown" xsi:type="array">
            <item name="path" xsi:type="string">Vendor_Module/js/flipdown-component</item>
            <item name="import" xsi:type="array">
                <item name="flipdown" xsi:type="string">Vendor_Module/js/flipdown.min.js</item>
            </item>
            <!-- Upcoming dynamic js feature -->
            <item name="load" xsi:type="array">
                <item name="onReveal" xsi:type="array">
                    <item name="0" xsi:type="string">[data-mage-init*="flipdown"]</item>
                </item>
            </item>
          </item>
        </item>
      </item>
    </argument>
  </arguments>
</referenceBlock>
```
